### PR TITLE
Fix #55, volume division

### LINKED ISSUE #55: Divide volume depending of number of points in an element

### PR DESCRIPTION
**Pull request by ezrayst**

### CODE REVIEW

File: include/mesh.tcc [hunk 1]
```diff
@@ -152,10 +169,12 @@ void Mesh<Tdim, Tvertices>::write_volumes(
     //! Write the total number of volumes (same as number of material points)
     volume_file << npoints_ << "\n";
 
-    //! Write element id and volumet
-    for (const auto& element : elements_) {
-      volume_file << id << '\t' << element->calculate_volume() << '\n';
-      ++id;
+    //! Write material point id and volume
+    for (const auto& materialpoint : materialpoints_) {
+      for (const auto& volume : materialpoint->volume()) {
```
Comment (kks32) [inline include/mesh.tcc:174]: Why are we iterating through volumes? Shouldn't each material point have it's own volume?

File: include/material_points.h [hunk 2]
```diff
@@ -42,6 +42,9 @@ class MaterialPoints {
   //! Return vector of stresses
   std::vector<Eigen::VectorXd> stress();
 
+  //! Return vector of volumes
```
Comment (kks32) [inline include/material_points.h:45]: Why is this a vector?

**Review by kks32 — CHANGES_REQUESTED** [review]
I don't understand the reasoning behind using a vector of volumes for each material point.

File: include/mesh.tcc (hunk 1 above)
Comment (ezrayst) [reply include/mesh.tcc:174]: Every `material point` have `point`, and `mesh` does not have access to `point`. Thus even though each `point` has its own `volume`, we need to iterate through vector of volumes from the output in `material point`. Remember that `material point` is a collection of point.

File: tests/point_test.cc [hunk 3]
```diff
@@ -20,8 +20,9 @@ TEST_CASE("Point base is checked in 2D", "[Point][2D]") {
     coordinate << 0, 0;
     Eigen::VectorXd stress(DIM * 2);
     stress << 0, 0, 0, 0;
+    const double volume = 0;
     auto node = std::unique_ptr<Point<DIM>>(
-        new Point<DIM>(index, global_index, coordinate));
+        new Point<DIM>(index, global_index, coordinate, volume));
```
Comment (kks32) [inline tests/point_test.cc:25]: Can we have two constructors with and without volume?

File: include/point.h [hunk 4]
```diff
@@ -13,10 +13,12 @@ class Point {
   //! Constructor with global id, id and coordinates
   //! \param[in] id index of the vertex
   //! \param[in] coord Coordinates of the point
```
Comment (kks32) [inline include/point.h:15]: Needs volume in comments.

**Review by kks32 — CHANGES_REQUESTED** [review]

**Review by kks32 — APPROVED** [review]

**Pull request merged**